### PR TITLE
ImportActivity intent filters.

### PR DESCRIPTION
**Pull request by rgmf**
These intent filters work for me. These are like the ones you can see in OsmAnd.

Also, read this: https://stackoverflow.com/a/8599921/8727038

### CODE REVIEW

**Comment by pstorch** [comment]
With this change at least open GPX files from FileManager is possible with OpenTracks, but still not sharing of GPX Tracks via Locus Maps nor OsmAnd.

**Comment by pstorch** [comment]
@rgmf @dennisguse I've got Locus and OsmAnd sharing of GPX tracks working with OpenTracks. Added my changes to this branch.

Finally I found out to listen to the SEND Action with the mime types of GPX and also read the ClipData of the Intent.
Maybe we should add similar Intent filters with mime types of kmz and kml as well.

**Comment by pstorch** [comment]
And we need the Intent filters with and without the mimetype. A null mimetype matches differently to a "" mimetype.

http://stackoverflow.com/questions/1733195/android-intent-filter-for-a-particular-file-extension

**Comment by rgmf** [comment]
Great @pstorch 👏 

> Maybe we should add similar Intent filters with mime types of kmz and kml as well.

I think we should.

File: src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -72,34 +72,94 @@ limitations under the License.
...
+                <data android:pathPattern=".*\\..*\\..*\\..*\\..*\\..*\\..*\\..*\\..*\\..*\\.gpx" />
             </intent-filter>
 
             <intent-filter>
+                <action android:name="android.intent.action.SEND"/>
+                <category android:name="android.intent.category.DEFAULT"/>
+                <data android:mimeType="application/gpx+xml"/>
+            </intent-filter>
+            <intent-filter>
+                <action android:name="android.intent.action.SEND"/>
```
Comment (dennisguse) [inline src/main/AndroidManifest.xml:126]: Why not include BROWSABLE and VIEW here?

**Review by dennisguse — APPROVED** [review]
Ugly ugly ugly, but if it works :)

File: src/main/AndroidManifest.xml (hunk 1 above)
Comment (pstorch) [reply src/main/AndroidManifest.xml:126]: No idea, I don't know what other apps are using. But if it doesn't harm, we could/should add it.

And yes, the whole intent filter thingy from Android is very ugly.

Comment (rgmf) [reply src/main/AndroidManifest.xml:126]: If you include VIEW together SEND a "Missing URL" error is gotten. So if we need VIEW and SEND then we have to create two intent-filter (the other solution is even more ugly: add ```tools:ignore="AppLinkUrlError"``` attribute to intent-filter).

**Comment by rgmf** [comment]
For me is okay (really ugly but it works).
@pstorch when you think is ready, splash the two commits and merge it.

**Comment by pstorch** [comment]
I added some more filters and also the mime types for kml and kmz, but haven't found another app which could share kml/kmz files. If you have some recommendations I would be happy to get a hint.
I'll merge the current status as it is an improvement anyway. If we find more cases, we can still further improve it.

**Pull request merged**